In [1]:
from keras.datasets import mnist

In [2]:
(train_images,train_labels),(test_images,test_labels) = mnist.load_data()

# Initialize models and Layers from keras

In [3]:
from tensorflow.keras import models
from tensorflow.keras import layers

In [4]:
model = models.Sequential([
    layers.Dense(512,activation='relu'),
    layers.Dense(10,activation='softmax')
])

In [5]:
model.compile(optimizer='rmsprop',loss='sparse_categorical_crossentropy',metrics=['accuracy'])

In [6]:
train_images = train_images.reshape((60000,28*28))
train_images = train_images.astype('float32')/255
test_images = test_images.reshape((10000,28*28))
test_images = test_images.astype('float32')/255

In [7]:
model.fit(train_images,train_labels,epochs=5,batch_size=64)

Epoch 1/5
938/938 [==============================] - 3s 3ms/step - loss: 0.2211 - accuracy: 0.9350
Epoch 2/5
938/938 [==============================] - 3s 3ms/step - loss: 0.0899 - accuracy: 0.9726
Epoch 3/5
938/938 [==============================] - 3s 3ms/step - loss: 0.0624 - accuracy: 0.9815
Epoch 4/5
938/938 [==============================] - 3s 3ms/step - loss: 0.0459 - accuracy: 0.9866
Epoch 5/5
938/938 [==============================] - 3s 3ms/step - loss: 0.0356 - accuracy: 0.9895


# Test Predictions

In [8]:
test_loss,test_acc = model.evaluate(test_images,test_labels)

313/313 [==============================] - 1s 2ms/step - loss: 0.0722 - accuracy: 0.9808


With a simple model , We got 98% accuracy and this can be futhur improved by adding 2 layer architecture. By using different model and we can improve 